docker-entrypoint.sh

```diff
@@ -0,0 +1,217 @@
+        cp -rf /etc/zulip "$DATA_DIR/settings/etc-zulip"
+    fi
+    rm -rf /etc/zulip
+    ln -sfT "$DATA_DIR/settings/etc-zulip" /etc/zulip
+}
+configurationSetUp() {
+    if [ ! -e "/etc/zulip/zulip-secrets.conf" ] || ([ "$FORCE_FIRST_START_INIT" == "True" ] || [ "$FORCE_FIRST_START_INIT" == "true" ]); then
+        echo "Generating Zulip secrets ..."
+        /root/zulip/scripts/setup/generate_secrets.py
+        echo "Secrets generation succeeded."
+    else
+        echo "Secrets already generated."
+    fi
+    echo "Checking configuration ..."
+    set +e
+    su zulip -c "/home/zulip/deployments/current/manage.py checkconfig"
+    RET_CODE=$?
+    set -e
+    if [[ $RET_CODE != 0 ]]; then
+        echo "Zulip config check failed with code $RET_CODE."
+        exit "$RET_CODE"
+    fi
+    echo "Configuration check done."
+}
+linkCertificates() {
+    if [ ! -e "$DATA_DIR/certs/zulip.key" ] && [ ! -e "$DATA_DIR/certs/zulip.combined-chain.crt" ]; then
+        echo "No certificates in '$DATA_DIR/certs'. Please create certificates and copy them to the docker data volume. Exiting: 1."
+        exit 1
+    else
+        echo "Certificates already exist."
+    fi
+    echo "Linking the certs."
+    if [ ! -e "$DATA_DIR/certs/zulip.key" ]; then
+        echo "No zulip.key given in $DATA_DIR."
+        echo "Certificates configuration failed."
+        exit 1
+    fi
+    if [ ! -e "$DATA_DIR/certs/zulip.combined-chain.crt" ]; then
+        echo "No zulip.combined-chain.crt given in $DATA_DIR."
+        echo "Certificates configuration failed."
+        exit 1
+    fi
+    ln -sfT "$DATA_DIR/certs/zulip.key" /etc/ssl/private/zulip.key
+    ln -sfT "$DATA_DIR/certs/zulip.combined-chain.crt" /etc/ssl/certs/zulip.combined-chain.crt
+    echo "Certificates configuration succeeded."
+}
+setupDatabase() {
+    if [ ! -z "$DB_HOST" ]; then
+        DB_CONNECTION_TIMEOUT=45
+        export PGPASSWORD="$DB_PASS"
+        echo "Waiting for database server to allow connections ..."
+        while ! /usr/bin/pg_isready -h "$DB_HOST" -p "$DB_HOST_PORT" -U "$DB_USER" -t 1 >/dev/null 2>&1
+        do
+            DB_CONNECTION_TIMEOUT=$(expr $DB_CONNECTION_TIMEOUT - 1)
+            if [[ $DB_CONNECTION_TIMEOUT -eq 0 ]]; then
+                echo "Could not connect to database server. Exiting ..."
+                exit 1
+            fi
+            echo -n "."
+            sleep 1
+        done
+        echo "(Re)creating database structure ..."
+        psql -h "$DB_HOST" -U "$DB_USER" <<EOF || :
+CREATE USER $DB_USER;
+ALTER ROLE $DB_USER SET search_path TO zulip,public;
+CREATE DATABASE $DB_NAME OWNER=$DB_USER;
+EOF
+        psql -h "$DB_HOST" -U "$DB_USER" <<EOF || :
+CREATE SCHEMA $DB_SCHEMA AUTHORIZATION $DB_USER;
+CREATE EXTENSION IF NOT EXISTS tsearch_extras SCHEMA zulip;
+EOF
+        unset PGPASSWORD
+        echo "Database structure (re)created."
+    else
+        echo "No database information given. Skipping ..."
+    fi
+}
```

Comment:
Like in the `scripts/lib/install`, I'll have to add some code there, but that shouldn't be the problem.